docs/extend/api.md

```diff
@@ -12,348 +12,967 @@ To use the built-in REST API as part of an integration, see [Consuming the REST
+
+class LabelResource extends AbstractDatabaseResource
+{
+    public function type(): string
+    {
+        return 'labels';
+    }
+
+    public function model(): string
+    {
+        return Label::class;
+    }
+}
+```
+
+### Scoping Database Resources
+
+The `scope` method is used to apply a query scope to the model. This is useful for applying visibility scoping and ensures no data is returned that the actor should not have access to, including when the resource is a serialized relationship of another resource.
+
+```php
+use Tobyz\JsonApiServer\Context;
+use Illuminate\Database\Eloquent\Builder;
+
+public function scope(Builder $query, Context $context): void
+{
+    $query->whereVisibleTo($context->getActor());
+}
+```
+
+### Listing Resources
 
-- The path should either be `/prefix/{id}` for get/update/delete, or `/prefix` for list/create.
-- the prefix (`tags` in the example above) must correspond to the JSON:API model type. You'll also use this model type in your serializer's `$type` attribute, and when registering the frontend model (`app.store.models.TYPE = MODEL_CLASS`).
-- The methods must match the example above.
+The `Index` endpoint lists the model instances. 
 
-Also, remember that route names (`tags.index`, `tags.show`, etc) must be unique!
+```php
+public function endpoints(): array
+{
+    return [
+        Endpoint\Index::make(),
+    ];
+}
+```
+
+:::info
+
+Find out more about the listing endpoint in the underlying package's documentation: https://tobyzerner.github.io/json-api-server/list.html
 
 :::
 
-The `Flarum\Api\Controller` namespace contains a number of abstract controller classes that you can extend to easily implement your JSON-API resources.
+#### Pagination
 
-:::info [Flarum CLI](https://github.com/flarum/cli)
+You can paginate the resources being **listed** to by specifying the `limit` and `maxLimit` through the `paginate` method:
 
-You can use the CLI to automatically create your endpoint controllers:
-```bash
-$ flarum-cli make backend api-controller
+```php
+use Flarum\Api\Endpoint;
+
+public function endpoints(): array
+{
+    return [
+        Endpoint\Index::make()
+            ->paginate(20, 50), // these are the default values, so you may omit them.
+    ];
+}
+```
+
+#### Sorting
+
+You can specify sort columns through the `sorts` method. For example the following will permit two sorting options: `createdAt` and `-createdAt`:
+
+```php
+use Flarum\Api\Sort\SortColumn;
+
+public function sorts(): array
+{
+    return [
+        SortColumn::make('createdAt'),
+    ];
+}
+```
+
+You can specify the default sort through the `defaultSort` method on the `Index` endpoint:
+
+```php
+use Flarum\Api\Endpoint;
+
+public function endpoints(): array
+{
+    return [
+        Endpoint\Index::make()
+            ->defaultSort('-createdAt'),
+            ->paginate(),
+    ];
+}
+```
+
+#### Searching and Filtering
+
+Read our [searching and filtering](search.md) guide for more information!
```

Comment:
Unrelated to this PR, but I think docs.flarum.org might not be up to date

Reply:
no, because that's 1.x, and this is for the 2.x branch which is still not published until 2.0 is done ;) https://github.com/flarum/docs/pull/466